File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.py
class Solution:
    def minTimeToVisitAllPoints(self, x: List[List[int]]) -> int:
        return sum([max(abs(x[i-1][0] - x[i][0]), abs(x[i-1][1] - x[i][1])) for i in range(1, len(x))])

File: 1266-minimum-time-visiting-all-points/1266-minimum-time-visiting-all-points.cpp
class Solution {
public:
    int solve(vector<int>cur,vector<int>prev){
        int x = abs(cur[0]-prev[0]);
        int y = abs(cur[1]-prev[1]);
        return max(x,y);
    }
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        int time = 0;
        for(int i = 1;i < points.size();i++){
            time += solve(points[i],points[i-1]);
        }
        return time;
    }
};